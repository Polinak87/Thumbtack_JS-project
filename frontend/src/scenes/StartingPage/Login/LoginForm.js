import React from 'react';
import FormField from '../../../components/FormField';
import Button, { green, large } from '../../../components/Button';

export default function LoginForm({ onChange, onSubmit }) {
  return (
    <form className="" name="LoginForm" onSubmit={onSubmit}>
      <FormField
        type="email"
        name="email"
        placeholder="Your Email"
        onChange={onChange}
      />
      <FormField
        type="password"
        name="password"
        placeholder="Your Password"
        onChange={onChange}
      />
      <Button className={`${large} ${green}`} type="submit">
        Log in
      </Button>
    </form>
  )
}