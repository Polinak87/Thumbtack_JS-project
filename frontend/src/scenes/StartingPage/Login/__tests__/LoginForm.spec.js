import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import LoginForm from '../LoginForm';

afterEach(cleanup);

describe('LoginForm', () => {
  it('simulates filling out and submiting the form', () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const props = {
      onChange,
      onSubmit,
    }
    const { getByPlaceholderText, getByText } = render(<LoginForm {...props} />);
    const inputEmail = getByPlaceholderText('Your Email');
    const inputPassword = getByPlaceholderText('Your Password');

    fireEvent.change(inputEmail, { target: { value: 'hh@mail.ru' } });
    fireEvent.change(inputPassword, { target: { value: 'ggg' } });
    fireEvent.submit(getByText('Log in'));

    expect(onChange).toHaveBeenCalledTimes(2);
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(inputEmail.value).toEqual('hh@mail.ru');
    expect(inputPassword.value).toEqual('ggg');
  });

  it('renders correctly', () => {
    const { container } = render(<LoginForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
