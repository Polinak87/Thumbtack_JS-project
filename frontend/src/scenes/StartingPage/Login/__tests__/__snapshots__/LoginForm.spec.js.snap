// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  class=""
  name="LoginForm"
>
  <div
    class="field"
  >
    <div
      class="control"
    >
      <input
        class="input"
        name="email"
        placeholder="Your Email"
        required=""
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    class="field"
  >
    <div
      class="control"
    >
      <input
        class="input"
        name="password"
        placeholder="Your Password"
        required=""
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    class="button is-block is-large is-fullwidth is-success"
    type="submit"
  >
    Log in
  </button>
</form>
`;
